### ISSUE TITLE
Issue in Transcribe decode

### ISSUE THREAD

**Issue by seertenedos** [iyulab/lm-supply#4]
When using the transcribe nuget the models download correctly now the result is always 
```
{
    "Id": 0,
    "Start": 0,
    "End": 30,
    "Text": "[Transcription placeholder - implement proper decoder]",
    "AvgLogProb": null,
    "NoSpeechProb": null,
    "CompressionRatio": null,
    "Words": null,
    "Duration": 30
  },
  {
    "Id": 1,
    "Start": 30,
    "End": 60,
    "Text": "[Transcription placeholder - implement proper decoder]",
    "AvgLogProb": null,
    "NoSpeechProb": null,
    "CompressionRatio": null,
    "Words": null,
    "Duration": 30
  },
```

The audio in my case is chinese and i tried with both of the code blocks below to get the same output though if i did not set ZH it detected it as EN

```
TranscribeOptions _options = new TranscribeOptions() {Language = "zh", WordTimestamps = true};
await using var transcriber = await LocalTranscriber.LoadAsync("default");
await transcriber.WarmupAsync();
sw.Stop();
var result = await transcriber.TranscribeAsync(wavPath, _options);

//console log the properties of result
Console.WriteLine($"Transcription Result for wav: {wavPath} at {DateTime.Now}");
Console.WriteLine($"Language: {result.Language}");
Console.WriteLine($"Full Text: {result.Text}");
```
```
await using var transcriber = await LocalTranscriber.LoadAsync("default");
await transcriber.WarmupAsync();
sw.Stop();
var result = await transcriber.TranscribeAsync(wavPath);

//console log the properties of result
Console.WriteLine($"Transcription Result for wav: {wavPath} at {DateTime.Now}");
Console.WriteLine($"Language: {result.Language}");
Console.WriteLine($"Full Text: {result.Text}");
```
It seemed to roughly match the instructions in the wiki.  I am trying to get the Chinese audio out with the timestamps so i can compare it with the OCR results from screenshots to help detect issues inthe OCR results.

**Comment by iyulab-caveman** [comment]
## Fixed in v0.8.11

The Whisper decoder has been fully implemented. The transcriber now returns actual transcription results instead of placeholder text.

### Changes Made:
- **WhisperTokenizer**: New GPT-2 BPE tokenizer implementation supporting Whisper's special tokens (language tokens, timestamp tokens, SOT sequence)
- **WhisperDecoder**: Autoregressive greedy decoder with proper logits processing
- **Model Configuration**: Added `HiddenSize` property to all model configurations for correct encoder output dimensions
- **Initialization**: Tokenizer and decoder are now properly loaded during model initialization

### Files Added/Modified:
- `src/LMSupply.Transcriber/Decoding/WhisperTokenizer.cs` (new)
- `src/LMSupply.Transcriber/Decoding/WhisperDecoder.cs` (new)
- `src/LMSupply.Transcriber/Core/OnnxTranscriberModel.cs` (updated)
- `src/LMSupply.Transcriber/Models/DefaultModels.cs` (updated)
- `src/LMSupply.Transcriber/Models/TranscriberModelInfo.cs` (updated)

All 36 Transcriber tests pass.

**Comment by iyulab-caveman** [comment]
## Fixed: Merged Model Support

The `decoder_model_merged.onnx` now works correctly. The issue was that the merged model requires additional inputs that weren't being provided.

### Root Cause
The merged decoder model from `onnx-community/whisper-*` requires:
- `use_cache_branch` - Boolean tensor to control KV cache branching
- `past_key_values.*` - 24 KV cache tensors (6 layers × 2 types × 2 components)

### Fix Applied
`WhisperDecoder` now:
1. **Auto-detects merged models** by checking for `use_cache_branch` input
2. **Provides required inputs**:
   - `use_cache_branch = false` (initial inference mode)
   - Empty KV cache tensors with shape `[1, num_heads, 0, head_dim]`

### Changes
- `src/LMSupply.Transcriber/Decoding/WhisperDecoder.cs` - Added merged model support

### Testing
- All 36 Transcriber tests pass
- Model loading and warmup verified with `onnx-community/whisper-base`

Please test with your Chinese audio file and report back. Note: You may need to clear your model cache to ensure you have the latest code:

```csharp
// The model should work now without any code changes on your side
await using var transcriber = await LocalTranscriber.LoadAsync("default");
var result = await transcriber.TranscribeAsync(wavPath, new TranscribeOptions 
{ 
    Language = "zh",
    WordTimestamps = true 
});
```

**Comment by seertenedos** [comment]
i am still having issues and i am now using version 0.8.15 This is the wav file i am testing with. https://drive.google.com/file/d/1ygSnPd3Y0SBfE4p4vUSnmXKqKhpSc4hZ/view?usp=sharing

You can see below that i have word timestamps on and they are not working. Also a number of the times the text is giant 30 second blocks and other times it does break them up correctly.


<img width="1167" height="812" alt="Image" src="https://github.com/user-attachments/assets/4005f274-7c25-4a34-bf29-55ac7f95f1b9" />

I have also attached the actual result i got
[transcribeResults.json](https://github.com/user-attachments/files/24601123/transcribeResults.json)

Also this is what i get from whisper for the same file in python but it runs on the GPU where as yours while it detects CUDA and even if i set CUDA it still uses CPU.

[whisper fast python.json](https://github.com/user-attachments/files/24601168/whisper.fast.python.json)

that said the python used the large model vs yours on default so i will test that later today as well to see if that fixes it. don't expect it to fix the GPU usage issues but maybe word timestamps will work then? 

If you change to the large model it currently throws an exception
```
System.Net.Http.HttpRequestException: 'Failed to download 'encoder_model.onnx' from 'onnx-community/whisper-large-v3'. Status: Unauthorized'
```

**Comment by seertenedos** [comment]
i also tried setting the model to what looked like the large instead of using "large" i also tried "onnx-community/whisper-large-v3-ONNX" but that resulted in the error
```
Microsoft.ML.OnnxRuntime.OnnxRuntimeException: '[ErrorCode:RuntimeException] Exception during initialization: file_size: The system cannot find the file specified.: "C:\Users\xxxx\.cache\huggingface\hub\models--onnx-community--whisper-large-v3-ONNX\snapshots\main\encoder_model.onnx_data
```
